### ISSUE TITLE
Audit Triggers

### ISSUE THREAD

**Issue by Ryan2065** [Eph-It/Automation#8]
Originally, like a fool, I was going to update an audit table every time objects were edited. 

Then I was reading and saw the suggestion to have SQL triggers do all the auditing, and that made sense. 

So need a good "template" trigger for audit things and then implement it on all tables that make sense.

Current idea is to have a single audit table that says user x performed y action on z object. It wouldn't be difficult (at this point in time) to pivot to a full audit that also says what they changed, but I don't see the need to be that granular right now.

**Comment by Ryan2065** [comment]
Started working on this.

I think if I can get a good template trigger we can programmatically add this. I've used the naming of "{TableName}Id" for all Id columns and there's a LastModifiedBy for whoever edited it. So it should be easy to generate the Update and Insert triggers to satisfy the audit log.

My thought for all objects to delete them would be flip a "IsDeleted" column to true, then it is practically removed but not from SQL yet. Then we'll have cleanup stuff that actually removes stuff. The reason for this complexity are the FK references. I wouldn't want a script to be deleted and then have no ability to cross reference an existing job. So the thought is we only delete objects once their dependent objects are gone.

Long story short, I'm not sure that I need an "OnDelete" trigger since users cannot directly delete rows.
